bin/wmcoreD

```diff
@@ -320,18 +337,63 @@ def status(configFile):
             return
         compDir = config.section_(component).componentDir
         compDir = os.path.expandvars(compDir)
-        daemonXml = os.path.join(compDir, "Daemon.xml")
-        if not os.path.exists(daemonXml):
-            print("Component:%s Not Running" % component)
-            continue
-        daemon = Details(daemonXml)
-        if not daemon.isAlive():
-            print("Component:%s Not Running" % component)
-        else:
-            print("Component:%s Running:%s" % (component, daemon['ProcessID']))
+        checkProcessThreads(component, compDir)
 
     sys.exit(0)
 
+def checkProcessThreads(component, compDir):
+    """
+    Helper function to check process and its threads for their statuses
+    :param component: component name
+    :return: prints status of the component process and its threads
+    """
+    # check if component daemon exists
+    daemonXml = os.path.join(compDir, "Daemon.xml")
+    if not os.path.exists(daemonXml):
+        print("Component:%s Not Running" % component)
+        return
+    daemon = Details(daemonXml)
```

Comment:
I would suggest to be consistent on the parsing of the XML file, at least within these changes. Here you use the long implemented class to parse the XML, but at the top of this module you use `extractFromXML()`.

Reply:
switched to use extractFromXML